agent/consul/rpc.go

```diff
@@ -534,9 +535,8 @@ func canRetry(args interface{}, err error) bool {
 		return true
 	}
 
-	// If we are chunking and it doesn't seem to have completed, try again
-	intErr, ok := args.(error)
-	if ok && strings.Contains(intErr.Error(), ErrChunkingResubmit.Error()) {
+	// If we are chunking and it doesn't seem to have completed, try again.
+	if err != nil && strings.Contains(err.Error(), ErrChunkingResubmit.Error()) {
```

Comment:
This is the buf fix for the second bug I noticed.

Reply:
Is there a test case that exercises this ErrChunkingResubmit flow to verify this is a-ok?

Reply:
There must not be, because it was previously broken. I'm not sure how chunking works, I'll see if I can write a test for it.

Reply:
I'll work on a test in a follow up. I think there is more cleanup to do around `canRetry` to prevent similar bugs in the future.